### ISSUE TITLE
中文问题(bug in Chinese Language)

### ISSUE THREAD

**Issue by diamont1001** [cozmo/jsQR#37]
当我使用中文生成的一张二维码时，使用 `decodeQRFromImage` decode 出来的字符为乱码，二维码图片如下：

解析出来的结果为：`'ä½ å¥½'`，其实是 `utf-8` 的问题，需要转换成 `utf-16`。

解决办法：
```js
...
const decoded =jsQR.decodeQRFromImage(imageData.data, imageData.width, imageData.height); // decoded is `'ä½ å¥½'`
const chineseStr = utf8to16(decoded); // chineseStr is '你好'

function utf8to16(str) {
    var out, i, len, c;
    var char2, char3;
    out = '';
    len = str.length;
    i = 0;
    while (i < len) {
        c = str.charCodeAt(i++);
        switch (c >> 4) {

        case 0: case 1: case 2: case 3: case 4: case 5: case 6: case 7:
            // 0xxxxxxx
            out += str.charAt(i - 1);
            break;
        case 12: case 13:
            // 110x xxxx   10xx xxxx
            char2 = str.charCodeAt(i++);
            out += String.fromCharCode(((c & 0x1F) << 6) | (char2 & 0x3F));
            break;
        case 14:
            // 1110 xxxx  10xx xxxx  10xx xxxx
            char2 = str.charCodeAt(i++);
            char3 = str.charCodeAt(i++);
            out += String.fromCharCode(((c & 0x0F) << 12) |
            ((char2 & 0x3F) << 6) |
            ((char3 & 0x3F) << 0));
            break;
        }
    }
    return out;
}
```

**Comment by diamont1001** [comment]
oh! sorry, the qr code just as below, and the words in qr code is '你好' in Chinese.

![image](https://user-images.githubusercontent.com/7159888/34926293-4c5d393c-f9e9-11e7-8c2f-a42aa41ba5fe.png)

**Comment by cozmo** [comment]
Good news - We just published `v1.0.0` #40 which is a total refactor of the decode function, and appears to solve this issue.

![](https://cl.ly/1X0T1Z3Q2c1O/Screen%20Shot%202018-01-14%20at%209.19.39%20PM.png)

You can try here or by upgrading the library - https://cozmo.github.io/jsQR/ - let me know if this solves your issue

**Comment by diamont1001** [comment]
Okay, this problem has been solved, thanks
